src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -5241,6 +5241,149 @@
+     * <pre>
+     * CLASS_DEF -&gt; CLASS_DEF
+     * |--MODIFIERS -&gt; MODIFIERS
+     * |   |--LITERAL_PUBLIC -&gt; public
+     * |   `--LITERAL_SEALED -&gt; sealed
+     * |--LITERAL_CLASS -&gt; class
+     * |--IDENT -&gt; Shape
+     * |--PERMITS_CLAUSE -&gt; permits
+     * |   |--IDENT -&gt; Circle
+     * |   |--COMMA -&gt; ,
+     * |   |--IDENT -&gt; Square
+     * |   |--COMMA -&gt; ,
+     * |   `--IDENT -&gt; Rectangle
+     * `--OBJBLOCK -&gt; OBJBLOCK
+     *     |--LCURLY -&gt; {
+     *     `--RCURLY -&gt; }
+     * </pre>
+     *
+     * @see <a href="https://docs.oracle.com/en/java/javase/15/docs/specs/sealed-classes-jls.html">
+     * Java Language Specification, &sect;8.1.7</a>
+     * @see #MODIFIERS
+     *
+     * @since 8.42
+     */
+    public static final int LITERAL_SEALED =
+        GeneratedJavaTokenTypes.LITERAL_sealed;
+
+    /**
+     * The {@code permits} restricted identifier.  This element appears
+     * as part of a class or interface declaration.
+     *
+     * <p>For example:</p>
+     * <pre>
+     * public sealed class Shape permits Circle, Square, Rectangle { }
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * CLASS_DEF -&gt; CLASS_DEF
+     * |--MODIFIERS -&gt; MODIFIERS
+     * |   |--LITERAL_PUBLIC -&gt; public
+     * |   `--LITERAL_SEALED -&gt; sealed
+     * |--LITERAL_CLASS -&gt; class
+     * |--IDENT -&gt; Shape
+     * |--PERMITS_CLAUSE -&gt; permits
+     * |   |--IDENT -&gt; Circle
+     * |   |--COMMA -&gt; ,
+     * |   |--IDENT -&gt; Square
+     * |   |--COMMA -&gt; ,
+     * |   `--IDENT -&gt; Rectangle
+     * `--OBJBLOCK -&gt; OBJBLOCK
+     *     |--LCURLY -&gt; {
+     *     `--RCURLY -&gt; }
+     * </pre>
+     *
+     * @see <a href="https://docs.oracle.com/en/java/javase/15/docs/specs/sealed-classes-jls.html">
+     * Java Language Specification, &sect;8.1.7</a>
+     * @see #MODIFIERS
+     *
+     * @since 8.42
+     */
+    public static final int LITERAL_PERMITS =
+        GeneratedJavaTokenTypes.LITERAL_permits;
+
+    /**
+     * A permits clause.  A permits clause's children are a comma separated list of one or
+     * more identifiers.
+     *
+     * <p>For example:</p>
+     * <pre>
+     * public sealed class Shape permits Circle, Square, Rectangle { }
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * CLASS_DEF -&gt; CLASS_DEF
+     * |--MODIFIERS -&gt; MODIFIERS
+     * |   |--LITERAL_PUBLIC -&gt; public
+     * |   `--LITERAL_SEALED -&gt; sealed
+     * |--LITERAL_CLASS -&gt; class
+     * |--IDENT -&gt; Shape
+     * |--PERMITS_CLAUSE -&gt; permits
+     * |   |--IDENT -&gt; Circle
+     * |   |--COMMA -&gt; ,
+     * |   |--IDENT -&gt; Square
+     * |   |--COMMA -&gt; ,
+     * |   `--IDENT -&gt; Rectangle
+     * `--OBJBLOCK -&gt; OBJBLOCK
+     *     |--LCURLY -&gt; {
+     *     `--RCURLY -&gt; }
+     * </pre>
+     *
+     * @see <a href="https://docs.oracle.com/en/java/javase/15/docs/specs/sealed-classes-jls.html">
```

Comment:
1. Links can point to specific sections, like https://docs.oracle.com/en/java/javase/15/docs/specs/sealed-classes-jls.html#jls-8.1.7 If you want to point to whole chapter, reference to specific section in link name is redundant
2. It looks like link was copy-pasted for all new tokens, but sections are different for these tokens

Reply:
I did not include the direct section links due to line length. The section numbers are corrected, thanks.